pkg/server/plugin/keymanager/gcpkms/client_fake.go

```diff
@@ -324,11 +330,35 @@ func (k *fakeKMSClient) setIsKeyDisabled(ok bool) {
 	k.keyIsDisabled = ok
 }
 
-func (k *fakeKMSClient) setGetPublicKeyErr(fakeError error) {
+func (k *fakeKMSClient) setGetPublicKeySequentialErrs(fakeErrors ...error) {
 	k.mu.Lock()
 	defer k.mu.Unlock()
+	errsRing := ring.New(len(fakeErrors))
+	for i := 0; i < errsRing.Len(); i++ {
+		errsRing.Value = fakeErrors[i]
+		errsRing = errsRing.Next()
+	}
+
+	k.getPublicKeyErrsRing = errsRing
+}
+
+func (k *fakeKMSClient) nextGetPublicKeySequentialErr() error {
+	k.mu.RLock()
+	defer func() {
+		k.mu.RUnlock()
 
-	k.getPublicKeyErr = fakeError
+		k.mu.Lock()
+		if k.getPublicKeyErrsRing != nil {
+			k.getPublicKeyErrsRing = k.getPublicKeyErrsRing.Next()
+		}
+		k.mu.Unlock()
+	}()
+
+	if k.getPublicKeyErrsRing == nil || k.getPublicKeyErrsRing.Value == nil {
+		return nil
+	}
+
+	return k.getPublicKeyErrsRing.Value.(error)
```

Comment:
I think the function logic would be simpler if you just make the write lock at the beginning at the cost of a bit of efficiency with the fine grained read/write locking.

Reply:
Definitely! since this bit of efficiency is not that critical here, I can give it a try 😄

Reply:
7d12460